src/Orleans.Runtime/Streams/PersistentStream/PersistentStreamPullingAgent.cs

```diff
@@ -604,6 +604,48 @@ private async Task RunConsumerCursor(StreamConsumerData consumerData, IStreamFil
             }
         }
 
+        private IBatchContainer GetBatchForConsumer(IQueueCacheCursor cursor)
+        {
+            if (this.options.BatchContainerBatchSize == 1)
+            {
+                Exception ignore;
+
+                if (!cursor.MoveNext())
+                {
+                    return null;
+                }
+
+                return cursor.GetCurrent(out ignore);
+            }
+            else if (this.options.BatchContainerBatchSize > 1)
+            {
+                Exception ignore;
+                int i;
+                var batchContainers = new List<IBatchContainer>();
+
+                for (i = 0; i < this.options.BatchContainerBatchSize; i++)
+                {
+                    if (!cursor.MoveNext())
+                    {
+                        break;
+                    }
+
+                    batchContainers.Add(cursor.GetCurrent(out ignore));
+                }
+
+                if (i == 0)
+                {
+                    return null;
+                }
+
+                return new BatchContainerBatch(batchContainers);
+            }
+            else
+            {
+                throw new OrleansConfigurationException("StreamPullingAgentOptions.BatchContainerBatchSize is negative");
```

Comment:
Suggest validation code a configuration time over failure at runtime.

Reply:
Is this supposed to read "Suggest validation code at configuration time over failure at runtime"?